src/mscorlib/src/System/Threading/Thread.cs

```diff
@@ -356,10 +358,21 @@ private void Start(ref StackCrawlMark stackMark)
 
 
 #if FEATURE_CORECLR
-        internal ExecutionContext ExecutionContext
+        public ExecutionContext ExecutionContext
         {
-            get { return m_ExecutionContext; } 
-            set { m_ExecutionContext = value; }
+            get
+            {
+                if (this == CurrentThread && m_ExecutionContext == null)
```

Comment:
This is performance critical path for ASP.NET - I do not think you want to change it like this.